ansible_ai_connect/ai/api/model_pipelines/factory.py

```diff
@@ -0,0 +1,369 @@
+            "dummy": DummyCompletionsPipeline,
+            "bam": BAMCompletionsPipeline,
+            "ollama": OllamaCompletionsPipeline,
+        }
+        if not settings.ANSIBLE_AI_MODEL_MESH_API_TYPE:
+            raise ValueError(
+                f"Invalid model mesh client type: {settings.ANSIBLE_AI_MODEL_MESH_API_TYPE}"
+            )
+
+        try:
+            expected_pipeline = pipelines[settings.ANSIBLE_AI_MODEL_MESH_API_TYPE]
+        except KeyError:
+            logger.error(
+                "Unexpected ANSIBLE_AI_MODEL_MESH_API_TYPE value: "
+                f"'{settings.ANSIBLE_AI_MODEL_MESH_API_TYPE}'"
+            )
+            raise ValueError(
+                "Unexpected ANSIBLE_AI_MODEL_MESH_API_TYPE value: "
+                f"'{settings.ANSIBLE_AI_MODEL_MESH_API_TYPE}'"
+            )
+
+        if self._completions_pipeline is None:
+            self._completions_pipeline = expected_pipeline(
+                inference_url=settings.ANSIBLE_AI_MODEL_MESH_API_URL
+            )
+
+        return self._completions_pipeline
+
+    def _get_content_match_pipeline(self) -> ModelPipelineContentMatch:
+        if self._content_match_pipeline:
+            return self._content_match_pipeline
+
+        from ansible_ai_connect.ai.api.model_pipelines.bam.pipelines import (
+            BAMContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.dummy.pipelines import (
+            DummyContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.grpc.pipelines import (
+            GrpcContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.http.pipelines import (
+            HttpContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.llamacpp.pipelines import (
+            LlamaCppContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.ollama.pipelines import (
+            OllamaContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.wca.pipelines_dummy import (
+            WCADummyContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.wca.pipelines_onprem import (
+            WCAOnPremContentMatchPipeline,
+        )
+        from ansible_ai_connect.ai.api.model_pipelines.wca.pipelines_saas import (
+            WCASaaSContentMatchPipeline,
+        )
+
+        pipelines = {
+            "grpc": GrpcContentMatchPipeline,
+            "wca": WCASaaSContentMatchPipeline,
+            "wca-onprem": WCAOnPremContentMatchPipeline,
+            "wca-dummy": WCADummyContentMatchPipeline,
+            "http": HttpContentMatchPipeline,
+            "llamacpp": LlamaCppContentMatchPipeline,
+            "dummy": DummyContentMatchPipeline,
+            "bam": BAMContentMatchPipeline,
+            "ollama": OllamaContentMatchPipeline,
+        }
+        if not settings.ANSIBLE_AI_MODEL_MESH_API_TYPE:
```

Comment:
Do you plan to rename the variable later?

Reply:
See https://issues.redhat.com/browse/AAP-33633

I plan on moving the whole way we configure a "Model Pipeline" from env vars to (probably) JSON per feature.

So we can configure Completions to use WCA, Playbook Generation to use our own hosted model etc.